From b4ea26d6dd2b386fd4f6d20df542a15a8abb8489 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Wed, 4 May 2022 12:07:39 +0000
Subject: [PATCH] IWYU: returning unique_ptr of WebContentsViewDelegate require
 its declaration.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix build issue with libstdc++ version of std::unique_ptr, breaking
compilation of ContentBrowserClient converting from nullptr.

Fixes this compilation error:
    ../../content/public/browser/content_browser_client.cc:104:10:   required from here
    /usr/include/c++/11/bits/unique_ptr.h:83:23: error: invalid application of ‘sizeof’ to incomplete type ‘content::WebContentsViewDelegate’

Bug: 957519
Change-Id: Iebbd94866b8a1a5b3c2c5da3dadfbecff254d751
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3615914
Commit-Queue: José Dapena Paz <jdapena@igalia.com>
Reviewed-by: Alexander Timin <altimin@chromium.org>
Cr-Commit-Position: refs/heads/main@{#999342}
---
 content/public/browser/content_browser_client.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/content/public/browser/content_browser_client.cc b/content/public/browser/content_browser_client.cc
index a6783920888d2..984205b3d1bb4 100644
--- a/content/public/browser/content_browser_client.cc
+++ b/content/public/browser/content_browser_client.cc
@@ -45,6 +45,7 @@
 #include "content/public/browser/url_loader_request_interceptor.h"
 #include "content/public/browser/vpn_service_proxy.h"
 #include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_view_delegate.h"
 #include "content/public/common/alternative_error_page_override_info.mojom.h"
 #include "content/public/common/content_features.h"
 #include "content/public/common/url_utils.h"
